scipy/sparse/_sputils.py

```diff
@@ -154,7 +154,7 @@ def get_index_dtype(arrays=(), maxval=None, check_contents=False):
     int32min = np.int32(np.iinfo(np.int32).min)
     int32max = np.int32(np.iinfo(np.int32).max)
 
-    dtype = np.intc
+    dtype = np.int32
```

Comment:
I _think_ this is a valid change, because the codegen for sparse matrices now produces `npy_int32` and `npy_int64` for index arrays. It's a little hard to figure out if this is actually 100% safe though.

Reply:
I believe that `sparsetools` used C `int`s when it was still written in Cython (~10 years ago, it's C++ now).

Reply:
Thanks for the patch and review. Just letting you my head got covidised this week so I'll need some more time to test the patch.

Reply:
Ouch - hope you're feeling better soon @drew-parsons